ServiceControl documentation for how to retain data between instances (#7207)
Particular/docs.particular.net: servicecontrol/migrations/retain-data-between-instances.md

- ServiceControl instances are deployed on aging hardware and need to be moved
- ServiceControl instances are being migrated to a different kind of deployment, such as moving from on-premises to cloud-hosted servers or from ServiceControl Management Utility installation to containers
- ServiceControl instance upgrades are not possible

This document gives an overview of strategies for retaining ServiceControl data, and links to more specific information on how to accomplish data retention for each.

## Overview

ServiceControl data consists of audit and error message data, managed by [audit instances](/servicecontrol/audit-instances/) and [error instances](/servicecontrol/servicecontrol-instances/) respectively.

[Monitoring instances](/servicecontrol/monitoring-instances/) do not use a persistent data store, and therefore have no data to retain.

## Strategies

> [!NOTE]
> It is worth assessing whether audit and error message data retention is required. For scenarios where retaining audit and error message data is not required (e.g. transient or test environment data that does not merit effort to retain), these strategies are not necessary.

> 

Reviewer: I think it helps to state when to go for Backup/Restore versus Instance replacement.

Just guessing some aspects:

- Downtime
- Lift and shift
- Version upgrades
- Very long audit retention periods
- Virtualization difference (VM to container)
- OS difference (Windows 2019 to 2022)

---

> title: Retain ServiceControl data between instances

Author: I rewrote a bit of the document to make it more clearly focused on data retention strategies.  The intent of this is to fill a gap for users who may be looking for a way to retain data between ServiceControl instances but aren't immediately aware that instance replacement or database backup and restore can achieve that.

### Audit data

Retaining audit data involves:

1. Adding the new audit instance as a remote
2. Disabling audit queue ingestion on the old audit instance
3. Decommission of the old audit instance _**after**_ all audit information has expired

Follow the [detailed audit instance replacement process](/servicecontrol/migrations/replacing-audit-instances/) to achieve retention of this data.

### Error data

Retaining error data involves:

1. Disabling error message ingestion so that new error messages will be temporarily held in the error queue
2. Retrying or archiving any failed messages so that the old error instance contains only ephemeral data like heartbeats and custom checks
3. Creating a new error instance configured to use the same audit instance(s)

Follow the [detailed error instance replacement process](/servicecontrol/migrations/replacing-error-instances/) to achieve retention of this data.

## Alternative retention strategies

Data retention can also be achieved by [backing up the ServiceControl data](/servicecontrol/backup-sc-database.md) and restoring it to the new instance.

> [!WARNING]

> > The [restrictions](/servicecontrol/backup-sc-database.md#important-notes-and-restrictions) should be strongly considered before moving forward with this approach.

Reviewer 2: ```suggestion
> The [restrictions](/servicecontrol/backup-sc-database.md#important-notes-and-restrictions) must considered before moving forward with this approach.
```
